packages/manager/apps/pci-instances/src/pages/instances/instance/dashboard/hooks/useDashboardPolling.ts

```diff
@@ -0,0 +1,106 @@
+import { useQueryClient } from '@tanstack/react-query';
+import { useCallback } from 'react';
+import { useNotifications } from '@ovh-ux/manager-react-components';
+import { ApiError } from '@ovh-ux/manager-core-api';
+import { useTranslation } from 'react-i18next';
+import { TInstance } from '@/types/instance/entity.type';
+import { useProjectId } from '@/hooks/project/useProjectId';
+import {
+  shouldRetryAfter404Error,
+  useInstancesPolling,
+} from '@/data/hooks/instance/polling/useInstancesPolling';
+import { buildPartialInstance } from '@/data/hooks/instance/builder/instance.builder';
+import { updateDashboardCache } from './useDashboard';
+
+const getPartialDeletedInstance = (id: string) =>
+  buildPartialInstance({ id })
+    .with('actions', [])
+    .with('addresses', new Map())
+    .with('volumes', [])
+    .with('status', 'DELETED')
+    .with('task', {
+      isPending: false,
+      status: null,
+    })
+    .build();
+
+const getPartialInstance = (instance: TInstance) =>
+  buildPartialInstance({ id: instance.id })
+    .with('actions', instance.actions)
+    .with('status', instance.status)
+    .with('task', {
+      isPending: instance.task.isPending,
+      status: instance.task.status ?? null,
+    })
+    .build();
+
+export const useDashboardPolling = ({
+  region,
+  instanceId,
+  isPending,
+}: {
+  region: string;
+  instanceId: string;
+  isPending?: boolean;
+}) => {
+  const queryClient = useQueryClient();
+  const projectId = useProjectId();
+  const { clearNotifications, addSuccess } = useNotifications();
+  const { t } = useTranslation('actions');
+  const pendingTasks = isPending ? [{ region, instanceId }] : [];
+
+  const handlePollingSuccess = useCallback(
+    (instance?: TInstance) => {
+      if (!instance) return;
+      const { status, task } = instance;
+      const isDeleted = !task.isPending && status === 'DELETED';
+      const newInstance = isDeleted
+        ? getPartialDeletedInstance(instance.id)
+        : getPartialInstance(instance);
```

Comment:
I think there is some misconception here, either you are manipulating DTOs or you are manipulating entities

If you are manipulating DTOs the functions already exists so no need to add more
If you are manipulating entities then you should not change their properties with a builder, it should be handled by the map DTO -> entity